### PR TITLE
Feat [CHK-3522] Added integration test in code review pipeline

### PR DESCRIPTION
**Pull request by EmanueleBVtech**
depends on https://github.com/pagopa/pagopa-payment-wallet-scheduler-service/pull/6

#### List of Changes
- Added integration tests in code review pipeline

#### Motivation and Context
This is standard procedure for all pagoPA recent progects

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docker-local/wallet-events-producer-testing/app.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+async function main() {
+    const uri = 'mongodb://mongodb:27017/?replicaSet=rs0';
+    const client = new MongoClient(uri);
+
+    try {
+        await client.connect();
+
+        const database = client.db('payment-wallet');
+        const collection = database.collection('payment-wallets-log-events');
```
Comment (scaminati-bv) [inline docker-local/wallet-events-producer-testing/app.js:13]: In this project we read data from **payment-wallets** collection

File: docker-local/wallet-events-producer-testing/data.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+[
+  {
```
Comment (scaminati-bv) [inline docker-local/wallet-events-producer-testing/data.json:2]: This is a sample data for payment-wallets collection:

`{
  "_id": "00a06383-1495-4b90-88f6-80c5fecf554a",
  "userId": "928bbaed-f7c2-41f9-b47c-9bce088322d6",
  "status": "VALIDATED",
  "paymentMethodId": "148ff003-46a6-4790-9376-b0e057352e45",
  "contractId": "W171896465077471gI",
  "applications": [
    {
      "_id": "PAGOPA",
      "status": "ENABLED",
      "creationDate": "2024-06-21T10:10:50.749146096Z",
      "updateDate": "2024-06-21T10:10:50.749146096Z",
      "metadata": {
        "onboardByMigration": "2024-06-21T10:10:50.749146096Z"
      }
    }
  ],
  "details": {
    "type": "CARDS",
    "bin": "223059",
    "lastFourDigits": "4353",
    "expiryDate": "203204",
    "brand": "MASTERCARD",
    "paymentInstrumentGatewayId": "41795763745a7246744943674e6265776e6a50674c375555754447687534386b4a7574706a5975307030593d",
    "_class": "it.pagopa.wallet.documents.wallets.details.CardDetails"
  },
  "clients": {
    "IO": {
      "status": "ENABLED"
    }
  },
  "version": 2,
  "creationDate": "2024-06-21T10:10:50.802632659Z",
  "updateDate": "2024-06-21T10:34:18.850935581Z",
  "onboardingChannel": "IO",
  "_class": "it.pagopa.wallet.documents.wallets.Wallet"
}`

[Here](https://github.com/pagopa/pagopa-payment-wallet-scheduler-service/pull/4/files#diff-4145e993a33209413e12c383914fcf689f922f49bf03455e555657f9561a8ee2) you can see the query used for wallets search.
Currently the status of the search wallet is “VALIDATED”, for me it is recommended to also enter wallets in a different status to verify the validity of the query

File: docker-local/wallet-events-producer-testing/app.js (hunk 1 above)
Comment (EmanueleBVtech) [reply docker-local/wallet-events-producer-testing/app.js:13]: i see this even on main branch, are you sure?
image

Comment (scaminati-bv) [reply docker-local/wallet-events-producer-testing/app.js:13]: DB Name: payment-wallet
Collection name: payment-wallets

Comment (EmanueleBVtech) [reply docker-local/wallet-events-producer-testing/app.js]: done here 8afa8fc4433641cde88a4495e2b402b0e2202b30

File: docker-local/wallet-events-producer-testing/data.json (hunk 2 above)
Comment (EmanueleBVtech) [reply docker-local/wallet-events-producer-testing/data.json]: done here 8afa8fc4433641cde88a4495e2b402b0e2202b30

**Comment by CianoDanilo** [comment]
Are we sure the test is working as expected? Looking at the code review pipeline logs, I don't see the message read from the queue printed correctly and I see an error returned by the jq command.
Screenshot 2024-11-07 alle 17 59 42
Screenshot 2024-11-07 alle 17 59 30

**Review by CianoDanilo — APPROVED** [review]

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**